luxonis_train/nodes/heads/base_head.py

```diff
@@ -0,0 +1,77 @@
+from abc import abstractmethod
+from typing import Generic, Literal
+
+from luxonis_train.nodes.base_node import (
+    BaseNode,
+    ForwardInputT,
+    ForwardOutputT,
+)
+from luxonis_train.utils import deep_merge_dicts
+
+
+class BaseHead(
+    BaseNode[ForwardInputT, ForwardOutputT],
+    Generic[ForwardInputT, ForwardOutputT],
+):
+    """Base class for all heads in the model.
+
+    @type parser: str | None
+    @ivar parser: Parser to use for the head.
+    @type is_softmaxed: bool | None
+    @ivar is_softmaxed: Whether the head uses softmax or not.
+    """
+
+    parser: (
+        Literal[
```

Comment:
Let's make this of type `str | None`  to allow someone to register a custom head with a  custom named parser which they would implement themselves.